ext/js/background/backend.js

```diff
@@ -2273,26 +2286,28 @@ export class Backend {
      * @param {string} prefix
      * @param {string} extension
      * @param {number} timestamp
-     * @param {import('api').InjectAnkiNoteMediaDefinitionDetails} definitionDetails
+     * @param {import('api').InjectAnkiNoteMediaDefinitionDetails?} definitionDetails
      * @returns {string}
      */
-    _generateAnkiNoteMediaFileName(prefix, extension, timestamp, definitionDetails) {
+    _generateAnkiNoteMediaFileName(prefix, extension, timestamp, definitionDetails = null) {
```

Comment:
Accepting null here (and actually using it) means that different definitions could result with the same filename if they are looked up in the same  second, right? Can this cause problems?

Reply:
Hadn't considered that. If someone did

1. copy image A
2. add term A
3. copy image B
4. add term B

and 2-4 happened within the same second, image B would unexpectedly overwrite image A in anki's media folder. 

It's pretty hard to do it that fast, but a couple ways to prevent it:

- Add more digits to the timestamp
- Generate a UID and either append it, or replace the timestamp
- Use [deleteExisting](https://foosoft.net/projects/anki-connect/index.html#media-actions): false in the ankiconnect call. Anki would then not overwrite files on filename. Would require another AnkiConnect call to check if a file is already added.

Reply:
Hmm, I guess using a more precise timestamp is the simplest solution.

BTW, is there any point in keeping the definitionDetails input at all? Wouldn't it be simpler to just eliminate it instead of creating a branch in the logic?

Reply:
Yeah, I guess with the more precise timestamp, the term/reading isn't needed in any of the filenames.